REGRESSION(252534@main): [ macOS ] TestWebKitAPI.ServiceWorkers.ServiceWorkerAndCacheStorageSpecificDirectories is constantly timing out

szewai opened the pull request:
#### 0915e758ce09519ffc471d32a1504e23e75e7084
<pre>
REGRESSION(252534@main): [ macOS ] TestWebKitAPI.ServiceWorkers.ServiceWorkerAndCacheStorageSpecificDirectories is constantly timing out
<a href="https://bugs.webkit.org/show_bug.cgi?id=242864">https://bugs.webkit.org/show_bug.cgi?id=242864</a>
rdar://problem/97205394

Reviewed by Chris Dumez.

We didn&apos;t set resolved configuration correctly in WebsiteDataStore::resolveDirectoriesIfNecessary, and the directories
were not resolved.

* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::retrieveCacheStorageParameters):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resolveDirectoriesIfNecessary):

Canonical link: <a href="https://commits.webkit.org/252625@main">https://commits.webkit.org/252625@main</a>
</pre>

cdumez commented on Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
```diff
@@ -300,7 +300,7 @@ void WebsiteDataStore::resolveDirectoriesIfNecessary()
         m_resolvedConfiguration->setResourceLoadStatisticsDirectory(resolveAndCreateReadWriteDirectoryForSandboxExtension(m_configuration->resourceLoadStatisticsDirectory()));
     if (!m_configuration->privateClickMeasurementStorageDirectory().isEmpty())
         m_resolvedConfiguration->setPrivateClickMeasurementStorageDirectory(resolveAndCreateReadWriteDirectoryForSandboxExtension(m_configuration->privateClickMeasurementStorageDirectory()));
-    if (!m_configuration->serviceWorkerRegistrationDirectory().isEmpty() && m_resolvedConfiguration->serviceWorkerRegistrationDirectory().isEmpty())
```
Below, there are a few that make sure the destination is empty to avoid overwriting. Why is this right here but not below?

szewai replied:
It's wrong below; I will update.

szewai replied:
(m_resolvedConfiguration is initialized to be m_configuration, so there's no point checking if its directory is empty.)

cdumez reviewed (approved):
OK, assuming EWS is happy.

szewai committed:
REGRESSION(252534@main): [ macOS ] TestWebKitAPI.ServiceWorkers.ServiceWorkerAndCacheStorageSpecificDirectories is constantly timing out
https://bugs.webkit.org/show_bug.cgi?id=242864
rdar://problem/97205394

Reviewed by Chris Dumez.

We didn't set resolved configuration correctly in WebsiteDataStore::resolveDirectoriesIfNecessary, and the directories
were not resolved.

* Source/WebKit/UIProcess/Network/NetworkProcessProxy.cpp:
(WebKit::NetworkProcessProxy::retrieveCacheStorageParameters):
* Source/WebKit/UIProcess/WebsiteData/WebsiteDataStore.cpp:
(WebKit::WebsiteDataStore::resolveDirectoriesIfNecessary):

Canonical link: https://commits.webkit.org/252625@main
